mne/encoding/model.py

```diff
@@ -0,0 +1,238 @@
+"""Base and high-level classes for fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from ..externals.six import string_types
+
+
+class SampleMasker(object):
+    """Remove samples with a particular value before fitting.
+
+    Parameters
+    ----------
+    estimator : instance of sklearn-style estimator
+        The estimator to be called after samples are removed.
+    mask_val : float, int, nan | callable | None
+        The value that will be removed from the dataset. If a callable,
+        it should return a boolean mask of length n_samples that takes
+        X an array of shape (n_samples, n_features) as input.
+    samples_train : array, shape (n_samples,) | None
+        An optional array of indices for manually specifying which samples
+        to keep during training. Defaults to None.
+    samples_pred : array, shape (n_samples,) | None
+        An optional array of indices for manually specifying which samples
+        to keep during prediction. Defaults to None.
+    mask_condition : 'all', 'any'
+        Whether only 1 `mask_val` will trigger removal of a row, or if
+        all values per row must be `mask_val` for removal. Defaults to 'all'.
+    """
+    def __init__(self, estimator, mask_val=None, samples_train=None,
+                 samples_pred=None, mask_condition='all'):
+        if all(ii is not None for ii in [mask_val, samples_train]):
+            raise ValueError('Supply one of mask_val | samples_train, or both')
+        if samples_train is not None:
+            samples_train = np.asarray(samples_train).astype(int)
+        self.samples_train = samples_train
+        if samples_pred is not None:
+            samples_pred = np.asarray(samples_pred).astype(int)
+        self.samples_pred = samples_pred
+
+        self.mask_val = mask_val
+        if mask_condition not in ['all', 'any']:
+            raise ValueError('condition must be one of "all" or "any"')
+        self.mask_condition = mask_condition
+        self.est = _check_estimator(estimator)
```

Comment:
or base_estimator, I think this is the convention for meta object, check on sklearn.ensemble